pkgs/stdenv/generic/check-meta.nix

```diff
@@ -117,6 +116,16 @@ let
   showLicense = showLicenseOrSourceType;
   showSourceType = showLicenseOrSourceType;
 
+  hasWarning = attrs: (attrs.meta.issues or []) != [];
+
+  ignoreWarningsDefaultPredicate = x: builtins.elem (getName x) (config.ignoreWarningsPackages or []);
+  ignoreWarningsPredicate = x: (config.ignoreWarningsPredicate or ignoreWarningsDefaultPredicate) x;
```

Comment:
Please add to `pkgs/top-level/config.nix`, where you can also set the default and a description for the [config options reference in the manual](https://nixos.org/manual/nixpkgs/unstable/#opt-enableParallelBuildingByDefault).

Reply:
I've seen that file, but I got confused because I couldn't find `allowInsecurePredicate` and other related options in there.

Reply:
It somewhat new and not complete yet.

Reply:
I gave it my best, please take a look. The description of the default values is a bit hairy tbh, because the default is defined over at check-meta.nix

Reply:
You could move that default to the `config.nix` module. `config.ignoreWarningsPackages` exists there too.

Reply:
The problem is that the default value references the `getName` helper function from `check-meta.nix`. Of course I could duplicate/inline it there, but that would be kind of ugly.

Reply:
> `(config.ignoreWarningsPredicate or ignoreWarningsDefaultPredicate)`

On second reading, `ignoreWarningsPredicate` will always be `config.ignoreWarningsPredicate`, because having an option for it means that it always exists.

Your logic (including "Note that `ignoreWarningsPackages` is only checked if `ignoreWarningsPredicate` is not specified.) can be specified entirely in the option:

```nix
mkOption {
  default = ignoreWarningsDefaultPredicate;
}
```

> the getName helper function from check-meta.nix. Of course I could duplicate/inline it there, but

Would be good to have this in lib. That way users can also try it in the repl for troubleshooting.

Turns out `getName` already exists. `getNameAndVersion` would be more appropriate anyway.

Reply:
> Turns out getName already exists. getNameAndVersion would be more appropriate anyway.

I'm tending more towards not matching on the version, as this is not a sensible default in all contexts (see https://github.com/NixOS/nixpkgs/pull/177272#discussion_r895011652 above). But if I match on the `pname` only, this would require handling of packages that don't have that attribute set. Currently I just ignore them, but I'd like to have `pname` as a requirement for setting a warning to some package.

Reply:
> tending more towards not matching on the version

Ah, I assumed your use of the locally defined `getName` was intentional, but it seems like you want `lib.getName` after all.

The local `getName` should be renamed, but that may not be a priority for this pr.